src/sst_file_writer.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020 Lucjan Suski
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//`
+
+use crate::{ffi, ffi_util::to_cpath, Error, Options};
+
+use libc::{self, c_char, size_t};
+use std::{ffi::CString, marker::PhantomData, path::Path};
+
+/// SstFileWriter is used to create sst files that can be added to database later
+/// All keys in files generated by SstFileWriter will have sequence number = 0.
+pub struct SstFileWriter<'a> {
+    pub(crate) inner: *mut ffi::rocksdb_sstfilewriter_t,
+    // Options are needed to be alive when calling open(),
+    // so let's make sure it doesn't get, dropped for the lifetime of SstFileWriter
+    phantom: PhantomData<&'a Options>,
+}
+
+unsafe impl<'a> Send for SstFileWriter<'a> {}
+unsafe impl<'a> Sync for SstFileWriter<'a> {}
+
+pub struct EnvOptions {
```

Comment:
Description ?

Reply:
this one should actually private, there is no way to customize it - rocksdb hardcodes the constructor call everywhere. Ill just unpublish it